app/Http/Controllers/Service/Admin/VouchersController.php

```diff
@@ -47,20 +50,63 @@ public function create()
      */
     public function storeBatch(Request $request)
     {
-        $new_vouchers = [];
+        $newVouchers = [];
+
+        $sponsorIds = \App\Sponsor::all()->pluck('id')->toArray();
+
+        $voucherRules = [
+            'sponsor_id' => [
+                'required',
+                Rule::in($sponsorIds)
```

Comment:
I *think* there is laravel magic that you don't need to get the sponsor ids.

`sponsor_id => ['required', 'exists:sponsors,id'];` should do the trick.